lib/tests/misc.nix

```diff
@@ -505,6 +505,38 @@ runTests {
       };
     };
 
+  testFoldl'Empty = {
+    expr = foldl' (acc: el: abort "operation not called") 0 [ ];
```

Comment:
```suggestion
    expr = foldl' (abort "function must not be used") 0 [ ];
```

Reply:
Doesn't actually work because the primop forces the function argument

Reply:
Fair enough. No need to eta expand it in `lib` itself. That'd be quite slow, and a little weird.